### PR TITLE
feat(start_planner): add surround moving obstacle check

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

**This [PR](https://github.com/autowarefoundation/autoware_launch/pull/723) should be merged first!!!**

This commit introduces a new feature in the start_planner module for checking surrounding moving obstacles.
- It adds parameters to specify the search radius and threshold velocity for moving obstacles, along with flags to indicate which types of objects should be checked.
- The `noMovingObjectsAround` function has been added to filter dynamic objects within a certain radius based on their velocity.
  - If no moving objects are detected, the function returns true; otherwise, it returns false.
- This feature enhances the safety of the start_planner by ensuring that the path can't be approved while surrond moving obstacles exist.

## Tests performed

confirmed with psim that safe status becomes false 
※safety check against dynamic object is disabled in the scene
image

- [x] [PASS TIERIV INTERNAL SCEANRIOS](https://evaluation.tier4.jp/evaluation/reports/f4f1d596-2618-5b14-a477-13d6af39f626?project_id=prd_jt)

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]
LGTM! thanks for quick work!

File: planning/behavior_path_planner_common/src/utils/path_safety_checker/objects_filtering.cpp [hunk 1]
```diff
@@ -44,6 +44,16 @@ bool position_filter(
 
   return (backward_distance < dist_ego_to_obj && dist_ego_to_obj < forward_distance);
 }
+
+bool position_filter(
```
[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner_common/src/utils/path_safety_checker/objects_filtering.cpp:48]: thanks for the comment!
Let me think :thinking:

[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner_common/src/utils/path_safety_checker/objects_filtering.cpp:48]: @zulfaqar-azmi-t4 
Do you think I can define different function name for positon_filter?
If so, is it better to change other functions name like the other `position_filter` and `velocity_filter`?
Do you have any specific concept in this part?

[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner_common/src/utils/path_safety_checker/objects_filtering.cpp:48]: @kosuke55 
How about `is_within_circle`?

[the remark this answers is not in the document]
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner_common/src/utils/path_safety_checker/objects_filtering.cpp]: renamed in https://github.com/autowarefoundation/autoware.universe/pull/5782/commits/13e06e631d373ff1969da050761b9149df1be250

**Pull request merged**